src/Lykke.AzureStorage/Blob/AzureBlob.cs

```diff
@@ -229,6 +230,67 @@ public async Task<IEnumerable<string>> FindNamesByPrefixAsync(string container,
             return results;
         }
 
+        /// <inheritdoc />
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="container"/> is null or whitespace.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="prefix"/> is null or whitespace.</exception>
+        public async Task DeleteBlobsByPrefixAsync(string container, string prefix)
+        {
+            if (string.IsNullOrWhiteSpace(container))
+                throw new ArgumentNullException(nameof(container));
+
+            if (string.IsNullOrWhiteSpace(prefix))
+                throw new ArgumentNullException(nameof(prefix));
+
+            BlobContinuationToken continuationToken = null;
+            var containerRef = GetContainerReference(container);
+
+            do
+            {
+                var result = await containerRef.ListBlobsSegmentedAsync(prefix, true, BlobListingDetails.None, null,
+                    continuationToken, GetRequestOptions(), null);
+
+                continuationToken = result.ContinuationToken;
+
+                await Task.WhenAll(result.Results
+                    .Select(item => (item as CloudBlob)?.DeleteIfExistsAsync())
+                    .Where(task => task != null)
+                );
+
+            } while (continuationToken != null);
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="container"/> is null or whitespace.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="prefix"/> is null or whitespace.</exception>
+        public async Task<IEnumerable<string>> GetListOfBlobKeysByPrefixAsync(string container, string prefix, int? maxResultsCount = null)
+        {
+            if (string.IsNullOrWhiteSpace(container))
+                throw new ArgumentNullException(nameof(container));
+
+            if (string.IsNullOrWhiteSpace(prefix))
+                throw new ArgumentNullException(nameof(prefix));
+
+            BlobContinuationToken continuationToken = null;
+            var results = new List<string>();
+            var containerRef = GetContainerReference(container);
+
+            do
+            {
+                var result = await containerRef.ListBlobsSegmentedAsync(prefix, true, BlobListingDetails.None, maxResultsCount,
```

Comment:
If maxResultsCount != null then we can pass maxResultsCount.Value - results.Count as max count parameter, to avoid returning more than requested

Reply:
If we omit this check. And only pass ```maxResultsCount``` to ```ListBlobsSegmentedAsync```, then ```continuationToken``` would not be null for each iteration, and we would take ```maxResultsCount``` on each iteration.
 (maxResultsCount == null || results.Count < maxResultsCount.Value)
```

For example if we specify ```maxResultsCount``` = 1, and remove check in while, but don't touch ```continuationToken``` check:
```cs
continuationToken != null
```
On each iteration of while we would take 1 result. ```continuationToken``` won't be null, thus loop would continue and we will take 1 more result on next iteration until we retrieve all results one by one.
Thus checking is required.

Reply:
Got your point, changed.